janusgraph-core/src/main/java/org/janusgraph/util/system/ExecuteUtil.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.util.system;
+
+import org.apache.commons.lang.UnhandledException;
+import org.janusgraph.diskstorage.BackendException;
+import org.janusgraph.util.datastructures.BackendExceptionWrapper;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class ExecuteUtil {
+
+    public static void executeWithCatching(ExecuteExceptionallyFunction function, BackendExceptionWrapper backendExceptionWrapper){
+        try {
+            function.execute();
+        } catch (BackendException backendException){
+            if(backendExceptionWrapper.getBackendException() == null){
+                backendExceptionWrapper.setBackendException(backendException);
```

Comment:
So this backendExceptionWrapper only catches the first exception, right? Is this by intention? I thought you would collect all exception stack traces and print them out together.

Reply:
I made it just for simplicity but I see that it may make sense to collect all the exceptions history. I will change that.